c7n/resources/apigw.py

```diff
@@ -309,7 +313,193 @@ def augment(self, resources):
+
+        futures = {}
+        results = set()
+
+        with self.executor_factory(max_workers=2) as w:
+            tasks = []
+            for r in resources:
+                r_method_set = method_set
+                if method_set == 'all':
+                    r_method_set = r.get('resourceMethods', {}).keys()
+                for m in r_method_set:
+                    tasks.append((r, m))
+            for task_set in utils.chunks(tasks, 20):
+                futures[w.submit(
+                    self.process_task_set, client, task_set)] = task_set
+
+            for f in as_completed(futures):
+                task_set = futures[f]
+
+                if f.exception():
+                    self.manager.log.warning(
+                        "Error retrieving integrations on resources %s",
+                        ["%s:%s" % (r['restApiId'], r['path'])
+                         for r, mt in task_set])
+                    continue
+
+                for i in f.result():
+                    if self.match(i):
+                        results.add(i['resourceId'])
+                        resource_map[i['resourceId']].setdefault(
+                            ANNOTATION_KEY_MATCHED_INTEGRATIONS, []).append(i)
+
+        return [resource_map[rid] for rid in results]
+
+    def process_task_set(self, client, task_set):
+        results = []
+        for r, m in task_set:
+            try:
+                integration = client.get_integration(
+                    restApiId=r['restApiId'],
+                    resourceId=r['id'],
+                    httpMethod=m)
+                integration.pop('ResponseMetadata', None)
+                integration['restApiId'] = r['restApiId']
+                integration['resourceId'] = r['id']
+                integration['resourceHttpMethod'] = m
+                results.append(integration)
+            except ClientError as e:
+                if e.response['Error']['Code'] == 'NotFoundException':
+                    pass
+
+        return results
+
+
+@RestResource.action_registry.register('update-integration')
+class UpdateRestIntegration(BaseAction):
+    """Change or remove api integration properties based on key value
+
+    :example:
+
+    .. code-block: yaml
+
+        policies:
+          - name: enforce-timeout-on-api-integration
+            resource: rest-resource
+            filters:
+              - type: rest-integration
+                key: timeoutInMillis
+                value: 29000
+            actions:
+              - type: update-integration
+                patch:
+                  - op: replace
+                    path: /timeoutInMillis
+                    value: "3000"
+    """
+
+    schema = utils.type_schema(
+        'update-integration',
+        patch={'type': 'array', 'items': OP_SCHEMA},
+        required=['patch'])
+    permissions = ('apigateway:PATCH',)
+
+    def validate(self):
+        found = False
+        for f in self.manager.filters:
```

Comment:
please switch out to using `self.manager.iter_filters()` as it properly handles nested blocks of filters.

Reply:
Fixed with latest commit.